Backprop

Dieter103 opened the pull request.

Dieter103 committed:
v

Dieter103 committed:
Merge branch 'master' of https://github.com/Penchant/EvolutionaryAlgorithms into Backprop

Dieter103 committed:
Hey

Dieter103 committed:
Comments added

Penchant commented on src/Evolution.java:
```diff
@@ -38,6 +47,16 @@
         return parents;
     }
 
+    public  void selectNewPopulation() {
```
Javadoc son!

Penchant commented on src/Network.java:
```diff
@@ -112,6 +112,22 @@ public Network(final List<Integer> hiddenLayers, int dimension, boolean isRadial
         setNodeConnections();
     }
 
+    public Network(final List<Integer> hiddenLayers, int dimension, int outputDimension, int populationSize) {
```
Needs more javadoc sir

Penchant commented on src/Network.java:
```diff
@@ -323,8 +343,7 @@ public double getPercentCorrect(){
     /**
-     * Use forwardProp to get output layer // TODO: ??????
-     *
+     * Uses outputs to update weights through backpropagation
```
Thank the lord somebody updated that

Dieter103 committed:
Comments added

Dieter103 committed:
fixed sort to delete correct part of list

DanielChurch commented on src/Evolution.java:
```diff
@@ -20,6 +18,18 @@
+     * @param populationSize Number of individuals to have in population
+     * @param numOfChildren Number of children
+     */
+    public Evolution(final List<Integer> hiddenLayers, int dimension, int outputDimension, int populationSize, int numOfChildren){
```
#styleCop `) {`

DanielChurch commented on src/Evolution.java:
```diff
@@ -39,6 +49,19 @@
+    	for (int i = 0; i<population.size(); i++)
+    	sortedPop.add(i, population.get(i));
+
+		Collections.sort(sortedPop, Comparator.comparing(s -> s.percentCorrect));
```
spaces > tabs

will-merryman committed:
Add files via upload

I've added a function to update the evolution strategy matrix. I've also allowed the mutation to aneal for non GA and DE algorithms.

Penchant committed:
Select new pop fix

DanielChurch committed:
actually add files

DanielChurch committed:
fix nits

DanielChurch committed:
Merge branch 'Backprop' of https://github.com/Penchant/EvolutionaryAlgorithms into Backprop

DanielChurch committed:
Merge branch 'master' of https://github.com/Penchant/EvolutionaryAlgorithms into Backprop

DanielChurch reviewed (approved):
ahha i can mwhahaha

DanielChurch committed:
nits
